clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -14289,6 +14289,9 @@ def note_unsafe_count_attributed_pointer_argument_null_to_nonnull : Note<
 def warn_unsafe_single_pointer_argument : Warning<
   "unsafe assignment to function parameter of __single pointer type">,
   InGroup<UnsafeBufferUsage>, DefaultIgnore;
+def warn_unsafe_single_pointer_assignment : Warning<
+  "unsafe assignment to __single pointer type">,
```

Comment:
Overall looks good, but can we do something about these diagnostics to indicate _why_ the assignment in unsafe, or what would be considered a safe assignment?